mesonbuild/cmake/fileapi.py

```diff
@@ -257,6 +256,27 @@ def parse_project(pro: T.Dict[str, T.Any]) -> T.Dict[str, T.Any]:
 
             return pro_data
 
+        def find_target(cnf_data: T.Dict[str, T.Any], id: str) -> T.Optional[T.Any]:
```

Comment:
I think the convention is to use a leading underscore for private functions in Meson. Also, it looks like you need to to mark these as `@staticmethod`? You aren't taking `self` or `cls`

Reply:
Well, everything is a private method already. :P 

Narrower input types would also be great if possible, although I think that would be ultimately an improvement on some existing code. What does CMakeConfigurations contain? Well, currently only created by parse_project...

`T.Optional[T.Any]` is an odd-looking annotation, since Any includes Optional. I *think* it always has to be a dict compatible with tgt_data?

Reply:
Alright, I looked into it and there are basically 3 solutions:
1. Remove the dicts and directly use the final classes from common.py (`CMakeConfiguration`, `CMakeProject`, `CMakeTarget`).  Would be great, and not just for typing, but that's a lot of refactoring!
2. Add TypedDict, its a bit less effort but it would be a waste if *1.* is the proper solution
3. Keep a consistent way of describing the root dict like the other functions in the file; not great for typing but serves more as a kind of documentation

I'll push the latter for now, I hope it will be enough for this PR.

Reply:
Cool. Let's aim for 1 in the future if you feel inclined to do that. Refactors that add typing and increase code clarity make subsequent review much nicer